alvr/xtask/src/ci.rs

```diff
@@ -100,4 +104,8 @@ pub fn clippy_ci() {
     if !out.status.success() {
         panic!("ci clippy didn't exit with 0 code, propagating failure");
     }
+
+    if !messages.is_empty() {
+        panic!("ci clippy produced warnings");
+    }
```

Comment:
I'm pretty sure there's also other compiler messages than diagnostics (you'd have to check the rustc docs tho), but we should only cause a ci failure when a diagnostic is actually emitted.

Reply:
I checked and there's actually some, but they're not very frequent. Either way the diagnostic message filtering should be moved to the main filter (or a second stacked filter_map), as that'll fully ensure that causes no issues.